### PR TITLE
TMO Spectrometer Class in Spectrometer.py

### PR DESCRIPTION
**Pull request by Mbosum**
## Description

Added a class at the end of spectrometer.py to connect to motor PVs associated with the TMO Photon Spectrometer.
## Motivation and Context

A tmo_spectrometer class is necessary for creating an entry in the happi database and for connecting to a typhos GUI to control motion along the axes.

## How Has This Been Tested?

I tested the module with flake8 and pytest and both came back clean.

(pcds-4.2.0) psdev04:pcdsdevices bosum123$ flake8 spectrometer.py
(pcds-4.2.0) psdev04:pcdsdevices bosum123$

(pcds-4.2.0) psdev04:pcdsdevices bosum123$ pytest spectrometer.py
============================================================== test session starts ===============================================================
platform linux -- Python 3.9.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
PyQt5 5.12.3 -- Qt runtime 5.12.9 -- Qt compiled 5.12.9
rootdir: /reg/g/pcds/epics-dev/bosum123/hutch_python/pcdsdevices
plugins: benchmark-3.4.1, timeout-1.4.2, qt-3.3.0
collected 0 items

================================================================ warnings summary ================================================================
../../../../../../../../cds/group/pcds/pyps/conda/py39/envs/pcds-4.2.0/lib/python3.9/site-packages/past/builtins/misc.py:45
  /cds/group/pcds/pyps/conda/py39/envs/pcds-4.2.0/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import reload

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================================================== 1 warning in 3.48s ===============================================================
(pcds-4.2.0) psdev04:pcdsdevices bosum123$

I also tested whether I  can make a device with it to verify the PV and component names:

Screenshot (275)

Screenshot (277)

## Where Has This Been Documented?

https://confluence.slac.stanford.edu/pages/viewpage.action?spaceKey=L2SI&title=Spectrometer+Motion+System+and+Spectrometer+Design

There is a docstring in the class.

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [ ] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [ ] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: pcdsdevices/spectrometer.py [hunk 1]
```diff
@@ -288,3 +288,33 @@ class Mono(BaseInterface, GroupDevice):
     removed = False
     transmission = 1
     SUB_STATE = 'state'
+
+
+class TMO_Spectrometer(BaseInterface, GroupDevice):
```
Comment (ZLLentz) [inline pcdsdevices/spectrometer.py:293]: Small thing, but usually we don't have underscores in class names as per pep8: https://www.python.org/dev/peps/pep-0008/#class-names

**Comment by ZLLentz** [comment]
The pytest test you've run in the description doesn't actually do anything. Pytest is a module that runs tests that we write ourselves.

In general there's no real test to write for a PR like this- the only thing to check is the ipython check you've dutifully added to the description. Thanks for doing this test.

It seems like something is wrong though- you should definitely not see a TimeoutError if things are working properly. Are you sure that the PVs are correct? Or is the IOC just down?

**Comment by ZLLentz** [comment]
Looking more closely, I think the PV names are wrong:
`TMO:SPEC::MMS:01`

**Comment by Mbosum** [comment]
Hi Zach, in the ipython test it looks like the PV prefix I was passing had a typo; fixed that now.

```
Python 3.9.6 | packaged by conda-forge | (default, Jul 11 2021, 03:39:48)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.26.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from pcdsdevices.spectrometer import TMO_Spectrometer

In [2]: tmo_spec = TMO_Spectrometer("TMO:SPEC", name="tmo_spec")

In [3]: tmo_spec.component_names
Out[3]:
('lens_x',
 'foil_x',
 'zone_plate_x',
 'zone_plate_y',
 'zone_plate_z',
 'yag_x',
 'yag_y',
 'yag_z',
 'yag_theta')

In [4]: tmo_spec.wait_for_connection()
---------------------------------------------------------------------------
TimeoutError                              Traceback (most recent call last)
<ipython-input-4-98d6a242cd1b> in <module>
----> 1 tmo_spec.wait_for_connection()

/cds/group/pcds/pyps/conda/py39/envs/pcds-4.2.0/lib/python3.9/site-packages/ophyd/device.py in wait_for_connection(self, all_signals, timeout)
   1151                                 for obj, description in funcs.items())
   1152             reasons.append(f'Pending operations: {pending}')
-> 1153         raise TimeoutError('; '.join(reasons))
   1154
   1155     def get_instantiated_signals(self, *, attr_prefix=None):

TimeoutError: Failed to connect to all signals: tmo_spec.user_readback (TMO:SPEC:MMS:01.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:01.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:01.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:01.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:01.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:01.SET), tmo_spec.velocity (TMO:SPEC:MMS:01.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:01.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:01.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:01.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:01.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:01.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:01.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:01.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:01.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:01.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:01.STOP), tmo_spec.disabled (TMO:SPEC:MMS:01.DISP), tmo_spec.description (TMO:SPEC:MMS:01.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:01.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:01:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:01:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:01:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:01:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:01:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:01.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:02.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:02.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:02.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:02.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:02.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:02.SET), tmo_spec.velocity (TMO:SPEC:MMS:02.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:02.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:02.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:02.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:02.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:02.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:02.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:02.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:02.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:02.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:02.STOP), tmo_spec.disabled (TMO:SPEC:MMS:02.DISP), tmo_spec.description (TMO:SPEC:MMS:02.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:02.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:02:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:02:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:02:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:02:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:02:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:02.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:03.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:03.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:03.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:03.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:03.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:03.SET), tmo_spec.velocity (TMO:SPEC:MMS:03.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:03.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:03.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:03.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:03.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:03.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:03.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:03.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:03.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:03.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:03.STOP), tmo_spec.disabled (TMO:SPEC:MMS:03.DISP), tmo_spec.description (TMO:SPEC:MMS:03.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:03.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:03:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:03:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:03:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:03:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:03:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:03.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:04.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:04.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:04.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:04.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:04.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:04.SET), tmo_spec.velocity (TMO:SPEC:MMS:04.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:04.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:04.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:04.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:04.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:04.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:04.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:04.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:04.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:04.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:04.STOP), tmo_spec.disabled (TMO:SPEC:MMS:04.DISP), tmo_spec.description (TMO:SPEC:MMS:04.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:04.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:04:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:04:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:04:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:04:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:04:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:04.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:05.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:05.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:05.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:05.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:05.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:05.SET), tmo_spec.velocity (TMO:SPEC:MMS:05.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:05.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:05.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:05.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:05.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:05.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:05.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:05.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:05.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:05.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:05.STOP), tmo_spec.disabled (TMO:SPEC:MMS:05.DISP), tmo_spec.description (TMO:SPEC:MMS:05.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:05.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:05:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:05:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:05:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:05:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:05:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:05.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:06.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:06.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:06.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:06.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:06.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:06.SET), tmo_spec.velocity (TMO:SPEC:MMS:06.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:06.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:06.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:06.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:06.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:06.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:06.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:06.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:06.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:06.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:06.STOP), tmo_spec.disabled (TMO:SPEC:MMS:06.DISP), tmo_spec.description (TMO:SPEC:MMS:06.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:06.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:06:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:06:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:06:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:06:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:06:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:06.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:07.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:07.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:07.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:07.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:07.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:07.SET), tmo_spec.velocity (TMO:SPEC:MMS:07.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:07.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:07.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:07.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:07.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:07.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:07.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:07.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:07.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:07.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:07.STOP), tmo_spec.disabled (TMO:SPEC:MMS:07.DISP), tmo_spec.description (TMO:SPEC:MMS:07.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:07.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:07:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:07:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:07:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:07:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:07:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:07.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:08.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:08.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:08.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:08.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:08.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:08.SET), tmo_spec.velocity (TMO:SPEC:MMS:08.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:08.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:08.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:08.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:08.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:08.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:08.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:08.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:08.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:08.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:08.STOP), tmo_spec.disabled (TMO:SPEC:MMS:08.DISP), tmo_spec.description (TMO:SPEC:MMS:08.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:08.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:08:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:08:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:08:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:08:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:08:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:08.SPMG), tmo_spec.user_readback (TMO:SPEC:MMS:09.RBV), tmo_spec.user_setpoint (TMO:SPEC:MMS:09.VAL), tmo_spec.user_offset (TMO:SPEC:MMS:09.OFF), tmo_spec.user_offset_dir (TMO:SPEC:MMS:09.DIR), tmo_spec.offset_freeze_switch (TMO:SPEC:MMS:09.FOFF), tmo_spec.set_use_switch (TMO:SPEC:MMS:09.SET), tmo_spec.velocity (TMO:SPEC:MMS:09.VELO), tmo_spec.acceleration (TMO:SPEC:MMS:09.ACCL), tmo_spec.motor_egu (TMO:SPEC:MMS:09.EGU), tmo_spec.motor_is_moving (TMO:SPEC:MMS:09.MOVN), tmo_spec.motor_done_move (TMO:SPEC:MMS:09.DMOV), tmo_spec.high_limit_switch (TMO:SPEC:MMS:09.HLS), tmo_spec.low_limit_switch (TMO:SPEC:MMS:09.LLS), tmo_spec.high_limit_travel (TMO:SPEC:MMS:09.HLM), tmo_spec.low_limit_travel (TMO:SPEC:MMS:09.LLM), tmo_spec.direction_of_travel (TMO:SPEC:MMS:09.TDIR), tmo_spec.motor_stop (TMO:SPEC:MMS:09.STOP), tmo_spec.disabled (TMO:SPEC:MMS:09.DISP), tmo_spec.description (TMO:SPEC:MMS:09.DESC), tmo_spec.dial_position (TMO:SPEC:MMS:09.DRBV), tmo_spec.plc.status (TMO:SPEC:MMS:09:PLC:sErrorMessage_RBV), tmo_spec.plc.err_code (TMO:SPEC:MMS:09:PLC:nErrorId_RBV), tmo_spec.plc.cmd_err_reset (TMO:SPEC:MMS:09:PLC:bReset_RBV), tmo_spec.plc.cmd_home (TMO:SPEC:MMS:09:PLC:bHomeCmd_RBV), tmo_spec.plc.home_pos (TMO:SPEC:MMS:09:PLC:fHomePosition_RBV), tmo_spec.motor_spmg (TMO:SPEC:MMS:09.SPMG)

```




As for the timeout error, I've always gotten this when I've done this test. Not sure the implication but things have still worked despite it.

**Comment by ZLLentz** [comment]
The timeout error either means that "something" is not connected, or that "something" was slow to connect for some reason. If all the signals you care about are connected it's probably ok but it's a bit confusing. Are you sure everything is working? In other devices it runs with no incident.

**Comment by ZLLentz** [comment]
Perhaps you're running the test wrong- are you doing it on a server with access to the PVs, such as `psbuild-rhel7-01`?

**Comment by ZLLentz** [comment]
Running this myself, a subset of the PVs do not connect- the "PLC" ones, such as the error reset. I think this must be an IOC issue, perhaps some non-standard naming or something strange, but I think the Python definitions are correct.

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**